engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -119,22 +114,30 @@ func (jm *JobMaster) InitImpl(ctx context.Context) error {
 		return err
 	}
 	if err := jm.preCheck(ctx, jm.initJobCfg); err != nil {
-		return err
+		return jm.Exit(ctx, frameModel.WorkerStatus{Code: frameModel.WorkerStatusError}, err)
 	}
-	if err := jm.checkpointAgent.Create(ctx); err != nil {
+	if err := jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil); err != nil {
 		return err
 	}
-	return jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil)
+	return jm.checkpointAgent.Create(ctx)
 }
 
 // Tick implements JobMasterImpl.Tick
 func (jm *JobMaster) Tick(ctx context.Context) error {
 	jm.workerManager.Tick(ctx)
 	jm.taskManager.Tick(ctx)
-	return jm.messageAgent.Tick(ctx)
+	if err := jm.messageAgent.Tick(ctx); err != nil {
+		return err
+	}
+	if jm.isFinished(ctx) {
```

Comment:
does it happen after checkpointAgent.Remove? These dependencies are hard to track, I'm afraid we will break it in future modification